great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -1153,12 +1234,54 @@ def add_profiler(
                 configuration_key=config.name,
             )
 
-        response = profiler_store.set(key=key, value=config)
+        response = persistence_fn(key=key, value=config)
```

Comment:
The only difference between `add`, `update`, and `add_or_update` is the callable we invoke at the end of this method